docs/csharp/tutorials/string-interpolation.md

```diff
@@ -0,0 +1,113 @@
+
+Now that you've installed all the tools, create a new .NET Core
+application. To use the command line generator, execute the following command in your favorite shell:
+
+`dotnet new`
+
+This command will create a barebones .NET core project with `project.json` and `Program.cs` files. You will need to execute `dotnet restore` to restore the dependencies needed to compile this project.
+
+To execute the program, use `dotnet run`. You should see "Hello, World" output to the console.
+
+## Intro to String Interpolation
+
+With `String.Format`, you would specify "placeholders" in a string that would be replaced by the parameters following the string. For instance:
+
+[!code-csharp[String.Format example](../../../samples/snippets/csharp/concepts/new-in-6/string-interpolation.cs#StringFormatExample)]  
+
+That will output "My name is Matt Groves".
+
+In C# 6, Instead of using String.Format, you can tell C# that a string is interpolated by prepending it with the `$` symbol, and then using the variables directly in the string. For instance:
+
+[!code-csharp[Interpolation example](../../../samples/snippets/csharp/concepts/new-in-6/string-interpolation.cs#InterpolationExample)]  
+
+You don't have to use just variables. You can use any expression within the brackets. For instance:
+
+[!code-csharp[Interpolation expression example](../../../samples/snippets/csharp/concepts/new-in-6/string-interpolation.cs#InterpolationExpressionExample)]  
+
+Which would output:
+
+```
+This is line number 1
+This is line number 2
+This is line number 3
+This is line number 4
+This is line number 5
+```
+
+## How string interpolation works
+
+Behind the scenes, this string interpolation syntax gets translated into String.Format by the compiler. So, you can do the same type of stuff you've done before with String.Format.
+
+For instance, you can add padding and numeric formatting:
+
+[!code-csharp[Interpolation formatting example](../../../samples/snippets/csharp/concepts/new-in-6/string-interpolation.cs#InterpolationFormattingExample)]  
+
+The above would output something like:
+
+```
+998        5,177.67
+999        6,719.30
+1000       9,910.61
+1001       529.34
+1002       1,349.86
+1003       2,660.82
+1004       6,227.77
+```
+
+## Localization and Internationalization
+
+An interpolated string supports `IFormattable` and `FormattableString`, which can be useful for internationalization.
+
+By default, an interpolated string uses the current culture. To use a different culture, you could cast it as `IFormattable`
+
+For instance:
+
+[!code-csharp[Interpolation internationalization example](../../../samples/snippets/csharp/concepts/new-in-6/string-interpolation.cs#InterpolationInternationalizationExample)]  
+
+For localization, keep in mind that instead of `{0}` style placeholders, you are now storing variable names, which means they must match up with variable names at runtime. If a variable name is not found, then a runtime error will be generated.
```

Comment:
The errors you describe are for any interpolated string, not just those that are cast as IFormattableStrings.

I'd move this section above the section on localization.